pulser/simulation/simconfig.py

```diff
@@ -51,55 +54,59 @@ def __str__(self) -> str:
         s = ""
         s += "Options:\n"
         s += "----------       \n"
-        s += "noise types:       " + str(self.noise) + "\n"
-        s += "spam dictionary:   " + str(self.spam_dict) + "\n"
-        s += "temperature:       " + str(self.temperature) + "K" + "\n"
-        s += "number of runs:    " + str(self.runs) + "\n"
-        s += "samples per runs:  " + str(self.samples_per_run) + "\n"
-        s += "laser waist:       " + str(self.laser_waist) + "μm \n"
+        s += "Noise types:       " + str(self.noise) + "\n"
+        s += "Spam dictionary:   " + str(self.spam_dict) + "\n"
+        s += "Temperature:       " + str(self.temperature) + "K" + "\n"
+        s += "Number of runs:    " + str(self.runs) + "\n"
+        s += "Samples per runs:  " + str(self.samples_per_run) + "\n"
+        s += "Laser waist:       " + str(self.laser_waist) + "μm \n"
         s += "\n" + "Solver Options: \n"+str(self.solver_options)[10:-1]+"\n"
         return s
 
     @property
-    def noise(self) -> List[str]:
+    def noise(self) -> list[NOISE_TYPES]:
         return self._noise
 
     @noise.setter
-    def noise(self, noise_types: List[str]) -> None:
-        self._noise: List[str] = []
+    def noise(self, noise_types: abc.Sequence[NOISE_TYPES]) -> None:
+        self._noise: list[NOISE_TYPES] = []
```

Comment:
How are you supposed to use this? It is written as if you can split the adding of noise in two commands, in which case a `setter` is not the best option. Let's say I want to add `SPAM` and then `doppler`:
```
sim_config.noise = ["SPAM"]
sim_config.noise = ["doppler"]
```
See how this can be misleading? Intuitively, I would expect `noise` to be only `["doppler"]` after this.

Reply:
Well, this is what happens in this case ? If you want to add both `SPAM` and `doppler` you would do 
`sim_config.noise = ['SPAM', 'doppler]`.
In the above example, `sim_config.noise = ['doppler']`. The `add_noise` method isn't supposed to be used by the user, maybe I should add an underscore in front of it.

Reply:
You're right, I missed `self.noise` being reset to `[]` at the beginning. Still, this is an awkward interface, `setters` are more suited for setting immutable objects.
I need to clarify how the user is supposed to use this class. From what I can tell, they initialize it and then add all the noise at once, is that it? If that's the case, you won't need the setter, because they can just provide the noise types to `__init__`.

Reply:
Yes, the user can initialize the noise all at once, but I also wanted to keep some flexibility and allow for changes. That said, I use `setters` in my code for convenience for the user and to run some code while modifying a field. For example, using `SimConfig.temperature = 500` will also modify the `doppler_sigma` accordingly, since the latter depends on the temperature.